### ISSUE TITLE
VS: 'create compilation database' greyed out in VS2019

### ISSUE THREAD

**Issue by PhilipOakley** [CoatiSoftware/Sourcetrail#1173]
I have just update my VS install and Sourcetrail after a period of non-use. 
I now have Sourcetrail 2021.1.30 and Visual Studio 2019 Community 16.9.3.
I installed the Sourcetrail extension via the market place (the menu sequence is different to that advertised in the docs https://github.com/CoatiSoftware/Sourcetrail/issues/1172)

I have the source code of Git for Windows (GfW) which is the target for the compilation. GfW has changed from using a .sln project file to using a `cmake` file (see https://github.com/git-for-windows/git/tree/vs/master), which VS detects and will build from. 

I'm checked out at https://github.com/git-for-windows/git/commit/0489a04b6d,  VS did compile, & test, the revision.

However I cannot find any way to get Sourcetrail to integrate with the VS Code. I opened Sourcetrail first, then started VS, loading and build/rebuilding the project, and then tried to create a Sourcetrail project to follow it. 

This fails at the 'add source group' step as I can't manage to get a Compilation database created, either from the Sourcetrail button "Create Compilation Database" (the INFO message about 'requesting IDE to create Compilation Database via plug-in' is generated but never returns), or from the VS menus: Extension > Sourcetrail > Create Compilation Database (which is greyed out). The Sourcetrail 'Open Log Directory' works.

Any suggestions or help? 
How do I check that the plug-in is installed properly - how can I tell?
Does the use of a `cmake` build confuse the Sourcetrail integration?
Do I have a PEBCAK problem?

**Comment by mlangkabel** [comment]
I don't think that the SourcetrailExtension for VS supports cmake projects. You could try the [Clang Power Tools](https://clangpowertools.com/) extension. After installing that one, right-click your solution inside the Solution Explorer of VS and there should be a "Clang Power Tools -> Export Compilation Database" option. Using the exported CDB, just create a CDB based project in Sourcetrail. You should still be able to use the Sourcetrail Extension to synchronize ST and VS.

**Comment by PhilipOakley** [comment]
@mlangkabel Thank you for the suggestion. I loaded the extension into VS, but it (VS & Clang) is still not detecting the 'project'. I.e. the Clang Power Tools options aren't showing when I right click.

I did get the attached warning saying that the CMake support is still in beta
![image](https://user-images.githubusercontent.com/914343/114715248-ca1cbb80-9d2a-11eb-95cb-94291ffad7f7.png)
The latest Git for Windows project (being pure cmake), doesn't have any vcxproj associated with it, which could be part of the problem.

I did notice that VS Tools there is a "Generate Sourcetrail Database" option (2nd from top) but it doesn't appear to do anything except give a warning "Could not find selected project. Make sure...". 
![image](https://user-images.githubusercontent.com/914343/114716028-91311680-9d2b-11eb-8bfe-93bb54f33dc2.png)
The option may have been there all the time without me noticing.

I may try the [Creating traditional Visual Studio sln from CMake](https://isocpp.org/blog/2020/06/creating-traditional-visual-studio-sln-from-cmake) converter. 
Or may try other routes via the [GfW SDK](https://gitforwindows.org/#contribute), but I'm cautious about jumping between fails 😉 

The VS view
![image](https://user-images.githubusercontent.com/914343/114716831-5aa7cb80-9d2c-11eb-9610-6b771a0c899b.png)
if that's any help for suggestions.
